src/main.c

```diff
@@ -0,0 +1,216 @@
+//LAB 7 - CAN Bus
+
+#include <zephyr.h>
+#include <kernel.h>
+#include <sys/printk.h>
+#include <device.h>
+#include <drivers/can.h>
+#include <drivers/gpio.h>
+#include <sys/byteorder.h>
+#include <stdio.h>
+
+
+#define LED0_NODE DT_ALIAS(led0)
+#define LED0	DT_GPIO_LABEL(LED0_NODE, gpios)
+#define PIN0	DT_GPIO_PIN(LED0_NODE, gpios)
+#define FLAGS0	DT_GPIO_FLAGS(LED0_NODE, gpios)
+
+#define LED1_NODE DT_ALIAS(led1)
+#define LED1	DT_GPIO_LABEL(LED1_NODE, gpios)
+#define PIN1	DT_GPIO_PIN(LED1_NODE, gpios)
+#define FLAGS1	DT_GPIO_FLAGS(LED1_NODE, gpios)
+
+#define LED2_NODE DT_ALIAS(led2)
+#define LED2	DT_GPIO_LABEL(LED2_NODE, gpios)
+#define PIN2	DT_GPIO_PIN(LED2_NODE, gpios)
+#define FLAGS2	DT_GPIO_FLAGS(LED2_NODE, gpios)
+
+#define LED3_NODE DT_ALIAS(led3)
+#define LED3	DT_GPIO_LABEL(LED3_NODE, gpios)
+#define PIN3	DT_GPIO_PIN(LED3_NODE, gpios)
+#define FLAGS3	DT_GPIO_FLAGS(LED3_NODE, gpios)
+
+CAN_DEFINE_MSGQ(receive_q, 50);
+
+//message -1
+struct zcan_frame frame1 = 
+        {
+                .id_type = CAN_STANDARD_IDENTIFIER,
+                .rtr = CAN_DATAFRAME,
+                .id = 0x123,
+                .dlc = 8,
+                .data = {1,2,3,4,5,6,7,8}
+        };
+//message -2
+struct zcan_frame frame2 = 
+        {
+                .id_type = CAN_STANDARD_IDENTIFIER,
+                .rtr = CAN_DATAFRAME,
+                .id = 0x456,
+                .dlc = 8,
+                .data = {1,2,3}
+
+        };
+
+//filter-id=0x123
+const struct zcan_filter single_filter_1 =
+ {
+        .id_type = CAN_STANDARD_IDENTIFIER,
+        .rtr = CAN_DATAFRAME,
+        .id = 0x123,
+        .rtr_mask = 1,
+        .id_mask = CAN_STD_ID_MASK
+};
+
+//filter-id=0x456
+const struct zcan_filter single_filter_2 =
+ {
+        .id_type = CAN_STANDARD_IDENTIFIER,
+        .rtr = CAN_DATAFRAME,
+        .id = 0x456,
+        .rtr_mask = 1,
+        .id_mask = CAN_STD_ID_MASK
+};
+
+//filter-mask =0 (essentially ignores the id)
+const struct zcan_filter all_filter =
+ {
+        .id_type = CAN_STANDARD_IDENTIFIER,
+        .rtr = CAN_DATAFRAME,
+        .id = 0x123,
+        .rtr_mask = 0,
+        .id_mask = CAN_STD_ID_MASK
+};
+
+
+//CAN messages
+const struct device *can_dev;
+int ret;
+int ret1;
+int filter_id1;
+int filter_id2;
+
+//LED
+const struct device *led_dev;
+const struct device *led_dev2;
+const struct device *led_dev3;
+const struct device *led_dev4;
+bool led_status;
+
+
+//send a single message to bus
+void send_message(struct zcan_frame *frame)
```

Comment:
A lot of these would likely fit better in a library file rather than as part of main.